EIPS/eip-7702.md

```diff
@@ -71,9 +71,10 @@ At the start of executing the transaction, after incrementing the sender's nonce
 3. Add `authority` to `accessed_addresses` (as defined in [EIP-2929](./eip-2929.md).)
 4. Verify the code of `authority` is either empty or already delegated.
 5. Verify the nonce of `authority` is equal to `nonce`.
-6. Add `PER_EMPTY_ACCOUNT_COST - PER_AUTH_BASE_COST` gas to the global refund counter if `authority` exists in the trie.
-7. Set the code of `authority` to be `0xef0100 || address`. This is a delegation designation.
-8. Increase the nonce of `authority` by one.
+6. Verify the nonce of `authority` is less than `2**64 - 1`.
```

Comment:
I think we should do this check earlier for optimization. We can check that `nonce < 2**64-1` immediately. If this is the nonce reported in the tuple field then we can immediately skip this authorization. Note that if the nonce of the `authority` is `2**64-1` and the nonce field itself is not, then rule 5 will fail. I suggest we put this after rule 1.

This also optimizes a lookup in the state database, since we now don't have to pull `authority` out of the state.

(Also, thanks for opening this PR :smile: )

Reply:
Sounds good, moved it to step 2.